### PR TITLE
A0-3351: Add dev env for dashbord

### PR DESCRIPTION
**Pull request by Marcin-Radecki**
* Adds `VITE_DISABLE_DEVNET` , similarly to exsting envs which filters out Testnet and Mainnet endpoint
* Adds new workflow that deploys to Devnet on every push to `master`
* Change deploy to Testnet workflow so it triggers only manually from now
* Fixed correct Dockerfile (we omitted that in the previous big merge to upstream commit)

### CODE REVIEW

File: .github/actions/deploy/action.yaml [hunk 1]
```diff
@@ -25,7 +26,7 @@ runs:
         repository: Cardinal-Cryptography/aleph-apps
         token: ${{ inputs.github_token }}
         path: "aleph-apps"
-        ref: ${{ inputs.environment }}
+        ref: ${{ inputs.environment == 'devnet' && 'main' || inputs.environment }}
```
Comment (youPickItUp) [inline .github/actions/deploy/action.yaml:29]: `main` or `master`?

Comment (Kosiare) [reply .github/actions/deploy/action.yaml:29]: We definitely use `master` here @Marcin-Radecki

File: .github/workflows/build-and-deploy-to-devnet.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - name: GIT | Checkout
+        uses: actions/checkout@v3
+
+      - name: Build and push devnet image
+        uses: ./.github/actions/build-and-push
+        id: build-devnet
+        with:
+          environment: devnet
+          build_envs: 'VITE_DISABLE_MAINNET=1 VITE_DISABLE_TESTNET=1 VITE_DISABLE_FIAT=1'
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/workflows/build-and-deploy-to-devnet.yaml:24]: Right, let me fix that.

File: .github/actions/deploy/action.yaml (hunk 1 above)
Comment (Marcin-Radecki) [reply .github/actions/deploy/action.yaml:29]: But this is checkout `ref` for our apps repo that has `main` default branch. So it seems correct?

image

File: .github/actions/deploy/action.yaml [hunk 3]
```diff
@@ -50,4 +51,4 @@ runs:
         message: 'Updating aleph-zero-dashboard ${{ inputs.environment }} docker image tag: ${{ inputs.image_tag }}'
         add: '*.yaml'
         cwd: 'aleph-apps'
-        branch: ${{ inputs.environment }}
+        branch: ${{ inputs.environment == 'devnet' && 'main' || inputs.environment }}
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/actions/deploy/action.yaml:54]: See above

File: .github/workflows/build-and-deploy-to-devnet.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+name: Build, push and deploy Aleph Zero Dashboard to devnet
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/workflows/build-and-deploy-to-devnet.yaml:7]: Sure, done

File: .github/actions/deploy/action.yaml (hunk 1 above)
Comment (youPickItUp) [reply .github/actions/deploy/action.yaml:29]: sorry, I misled us, my bad :see_no_evil:

**Review by youPickItUp — APPROVED** [review]

**Pull request merged**